File: esthesis-platform-backend/esthesis-platform-backend-server/src/main/java/esthesis/platform/backend/server/dto/CampaignStatsDTO.java
package esthesis.platform.backend.server.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@NoArgsConstructor
@Accessors(chain = true)
public class CampaignStatsDTO {
  private List<Integer> groupMembers;
  private List<Integer> groupMembersReplied;
  private int allMembers;
  private int membersContacted;
  private int membersContactedButNotReplied;
  private int membersReplied;
  private int successRate;
  private int progress;
  private String duration;
  private String eta;
}

File: esthesis-ui/src/app/dto/campaign-stats-dto.ts
export interface CampaignStatsDto {
  groupMembers?: [number];
  groupMembersReplied?: [number];
  membersContacted?: number;
  membersContactedButNotReplied?: number;
  membersReplied?: number;
  successRate?: number;
  allMembers?: number;
  progress?: number;
  duration?: string;
  eta?: string;
}
